server/src/infra/repositories/filesystem.provider.ts

```diff
@@ -11,25 +11,16 @@ import { ImmichLogger } from '@app/infra/logger';
 import archiver from 'archiver';
 import chokidar, { WatchOptions } from 'chokidar';
 import { glob } from 'glob';
+import { Span } from 'nestjs-otel';
 import { constants, createReadStream, existsSync, mkdirSync } from 'node:fs';
-import fs, { copyFile, readdir, rename, stat, utimes, writeFile } from 'node:fs/promises';
+import fs from 'node:fs/promises';
 import path from 'node:path';
+import { DecorateAll } from '../infra.utils';
 
+@DecorateAll(Span())
 export class FilesystemProvider implements IStorageRepository {
   private logger = new ImmichLogger(FilesystemProvider.name);
 
-  readdir = readdir;
-
-  copyFile = copyFile;
-
-  stat = stat;
-
-  writeFile = writeFile;
-
-  rename = rename;
-
-  utimes = utimes;
```

Comment:
Is this considered bad practice?

Reply:
An earlier version of the PR added the decorator to every method, but you can't add decorators with this syntax so I refactored it. I think the class decorator *might* work with this syntax, but I'm not sure if these are actually considered methods and not bolted on functions.

I prefer the refactored version personally because it's easier to make changes to them.